### PR TITLE
Add ArrayList{Unmanaged}.fromOwnedSliceSentinel and ArrayListUnmanged.fromOwnedSlice

### LINKED ISSUE #14315: ArrayListUnmanaged needs fromOwnedSlice, ArrayList and ArrayListUnmanaged could use a fromOwnedSliceSentinel
### Zig Version

0.11.0-dev.1301+0e659042a

### Steps to Reproduce and Observed Behavior

Std lib enhancement

Look in `lib/std/array_list.zig` and fail to find a `fromOwnedSlice` function for ArrayListUnmanaged, be sad.

Also look for the sentinel equivalent `fromOwnedSliceSentinel` to match with `toOwnedSliceSentinel` in both ArrayList and ArrayListUnmanaged and continue to be sad.

### Expected Behavior

To have both of these functions available. Helps going back and forth with C libs that expect null terminators.

### PR DESCRIPTION
**Pull request by michaelbartnett**
Fixes #14315

### CODE REVIEW

File: lib/std/array_list.zig [hunk 1]
```diff
@@ -87,6 +87,17 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
             };
         }
 
+        /// ArrayList takes ownership of the passed in slice. The slice must have been
+        /// allocated with `allocator`.
+        /// Deinitialize with `deinit` or use `toOwnedSlice`.
+        pub fn fromOwnedSliceSentinel(allocator: Allocator, comptime sentinel: T, slice: [:sentinel]T) Self {
+            return Self{
+                .items = @ptrCast([*]T, slice.ptr)[0..slice.len],
```
Comment (silversquirl) [inline lib/std/array_list.zig:95]: What is this line trying to do? The `ptrCast` is a no-op (it's already `[*]T`) and the slice doesn't do anything. Did you mean to do `slice.ptr[0..slice.len + 1]`, which would extend the length to include the sentinel?

Comment (michaelbartnett) [reply lib/std/array_list.zig:95]: Mmm that's a good callout, the ptrCast is some sloppiness.

The intended outcome here is to have the items.len be the length of the slice _excluding_ the sentinel, but including the sentinel in the capacity.

So the .items initializer can just be:

```zig
                .items = slice[0..slice.len],
```

Comment (silversquirl) [reply lib/std/array_list.zig]: Oh, you don't need to do anything fancy then, just use `.items = slice`

The length does not include the sentinel

Comment (michaelbartnett) [reply lib/std/array_list.zig]: 🤦‍♂️ D'oh right, I should stop replying to PR reviews right before bed. Thank you 🙂

**Pull request merged**